backend/shelf-server/src/main/java/com/htec/shelfserver/service/RegisterService.java

```diff
@@ -0,0 +1,154 @@
+
+        String salt = tokenGenerator.generateSalt(8);
+        userEntity.setSalt(salt);
+
+        String encryptedPassword = bCryptPasswordEncoder.encode(userDTO.getPassword() + salt);
+        userEntity.setPassword(encryptedPassword);
+
+        UserEntity storedUser = userRepository.save(userEntity);
+        createAndSendToken(storedUser);
+
+    }
+
+    public void registerUserMicrosoft(String bearerToken) {
+
+
+        RestTemplate restTemplate = new RestTemplate();
+        String url = "https://graph.microsoft.com/v1.0/me";
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Authorization", "Bearer eyJ0eXAiOiJKV1QiLCJub25jZSI6InVkdi1yN3ZZYUJBbGZTS2s1T0pLa3RVUHdWbDFHUHY1RDk5bExwbW8xWm8iLCJhbGciOiJSUzI1NiIsIng1dCI6Imwzc1EtNTBjQ0g0eEJWWkxIVEd3blNSNzY4MCIsImtpZCI6Imwzc1EtNTBjQ0g0eEJWWkxIVEd3blNSNzY4MCJ9.eyJhdWQiOiJodHRwczovL2dyYXBoLm1pY3Jvc29mdC5jb20iLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC85Zjg1NjY1Yi03ZWZkLTQ3NzYtOWRmZS1iNmJmZGEyNTY1ZWUvIiwiaWF0IjoxNjM3OTIxODQzLCJuYmYiOjE2Mzc5MjE4NDMsImV4cCI6MTYzNzkyNjUxMywiYWNjdCI6MCwiYWNyIjoiMSIsImFpbyI6IkUyWmdZQ2lvNUhQWDJCblVkdjZQc1dxcGNmT2FEdm5rOTh6N3RoVzhOWjhzbi9vcnhBRUEiLCJhbXIiOlsicHdkIl0sImFwcF9kaXNwbGF5bmFtZSI6Ik9mZmljZUhvbWUiLCJhcHBpZCI6IjQ3NjU0NDViLTMyYzYtNDliMC04M2U2LTFkOTM3NjUyNzZjYSIsImFwcGlkYWNyIjoiMiIsImZhbWlseV9uYW1lIjoiTWFrb2pldmljIiwiZ2l2ZW5fbmFtZSI6Ikx1a2EiLCJpZHR5cCI6InVzZXIiLCJpcGFkZHIiOiI4Mi4xMTcuMjE3LjIyMiIsIm5hbWUiOiJMdWthIE1ha29qZXZpYyIsIm9pZCI6IjRlNGJkMWMxLWYzODQtNDJiZS05M2IxLTMyODBjMjhhODdlYyIsInBsYXRmIjoiOCIsInB1aWQiOiIxMDAzMjAwMUEwRkI1NDM4IiwicmgiOiIwLkFVNEFXMmFGbl8xLWRrZWRfcmFfMmlWbDdsdEVaVWZHTXJCSmctWWRrM1pTZHNwT0FMVS4iLCJzY3AiOiJlbWFpbCBGaWxlcy5SZWFkV3JpdGUuQWxsIE5vdGVzLkNyZWF0ZSBvcGVuaWQgUGVvcGxlLlJlYWQgUHJlc2VuY2UuUmVhZC5BbGwgcHJvZmlsZSBTaXRlcy5SZWFkLkFsbCBUYXNrcy5SZWFkV3JpdGUgVXNlci5SZWFkQmFzaWMuQWxsIiwic3ViIjoiRUhUcHNLcnI2RVdJSlA5RUloaWpZXzFTdmVuTnZsdmQ3VXpvSmV3eG10YyIsInRlbmFudF9yZWdpb25fc2NvcGUiOiJFVSIsInRpZCI6IjlmODU2NjViLTdlZmQtNDc3Ni05ZGZlLWI2YmZkYTI1NjVlZSIsInVuaXF1ZV9uYW1lIjoibHVrYS5tYWtvamV2aWNAaHRlY2dyb3VwLmNvbSIsInVwbiI6Imx1a2EubWFrb2pldmljQGh0ZWNncm91cC5jb20iLCJ1dGkiOiJoOVFkenhZU2RVdTNaNVFzeEdzVEFRIiwidmVyIjoiMS4wIiwid2lkcyI6WyJiNzlmYmY0ZC0zZWY5LTQ2ODktODE0My03NmIxOTRlODU1MDkiXSwieG1zX3N0Ijp7InN1YiI6InA2QjdwNjhOcER5TE14OTdHQlJvVkJraHJRdWh1VkVkUXdLRWlXQ1UxaG8ifSwieG1zX3RjZHQiOjE2MzQwMzk0NjF9.TtNhwRuT7RKH7nshPN5vb13jEdVDAvnElMViEmqBuXPVGE0ykjVsNaUO06vrTGy9aKP0MjmeJDyldHwsslqGOGhgfn5OYXnVvLFnjooP5-uUmjw1uqZePWPavE3WZuwtKFnR8Olr2o7g6YDUe8W0rzCObEieOS_MYI6iuUnoC18h85RCaupqtJAaSA30ESM84UMxucA3l-ABhRUUBgQq7Ah3mQzBM9fltWe_1HeBB2XJy01APsMsMwUOZYxa0xZ_nXiIvzN59807J_uy08FUgDmvlb89L_uHYDJbcgw5bXrBBO1rKX0zLvPB4MEmkL1RKfkJRxzJetDZdjmUELLddQ");
+
+
+        ResponseEntity<String> response = restTemplate.exchange(url, HttpMethod.GET, new HttpEntity<>(headers), String.class);
+
+        System.out.println(response);
+        //getForEntity( url, request , String.class );
+
+
+//            URL url = new URL("https://graph.microsoft.com/v1.0/me");
```

Comment:
try to avoid commented code, you can remove it if you need, git history will have it

Reply:
removed.